index.js

```diff
@@ -1,58 +1,45 @@
-var hyperlog = require('hyperlog')
-var fdstore = require('fd-chunk-store')
 var path = require('path')
 var mkdirp = require('mkdirp')
-var osmdb = require('osm-p2p-db')
-var once = require('once')
-
+var kappa = require('kappa-core')
+var osmdb = require('kappa-osm')
 var LevelUp = require('levelup')
 var leveldown = require('leveldown')
-var deflevel = require('deferred-leveldown')
-var deferredStore = require('deferred-chunk-store')
+var once = require('once')
+var Deflevel = require('deferred-leveldown')
+var raf = require('random-access-file')
 
 module.exports = function (opts) {
   if (typeof opts === 'string') opts = { dir: opts }
   if (!opts) opts = {}
-  var dir = opts.dir || path.resolve('./osm-p2p-data')
-  var defIndexDB = new deflevel()
-  var defLogDB = new deflevel()
+  var dir = opts.dir || path.resolve('./osm-p2p-kappa')
 
-  var log = hyperlog(levelup(defLogDB), { valueEncoding: 'json' })
-  var chunkLength = opts.chunkLength || 4096
-  var defStore = deferredStore(chunkLength)
+  var defIndexDB = new Deflevel()
+  var index = levelup(defIndexDB)
 
   mkdirp(dir, function (err) {
     if (err) osm.emit('error', err)
-    var store = fdstore(chunkLength, path.join(dir, 'kdb'))
-    defStore.setStore(store)
 
     var indexDown = leveldown(path.join(dir, 'index'))
     indexDown.open(function (err) {
       if (err) osm.emit('error', err)
       else defIndexDB.setDb(indexDown)
     })
 
-    var logDown = leveldown(path.join(dir, 'log'))
-    logDown.open(function (err) {
-      if (err) osm.emit('error', err)
-      else defLogDB.setDb(logDown)
-    })
-    db.close = function (cb) {
+    index.close = function (cb) {
       cb = once(cb || noop)
       var pending = 2
       indexDown.close(onclose)
-      logDown.close(onclose)
       function onclose (err) {
         if (err) cb(err)
         else if (--pending === 0) cb(null)
       }
     }
   })
-  var db = levelup(defIndexDB)
+
   var osm = osmdb({
-    log: log,
-    db: db,
-    store: defStore
+    core: kappa(dir, {valueEncoding: 'json'}),
+    index: index,
+    storage: raf
```

Comment:
is this the `random-access-*` module that we want to use?

Reply:
yep! that's the canonical on-disk module, which I think maps well onto the old on-disk level storage